app/account/services_views.py

```diff
@@ -0,0 +1,216 @@
+    Returns:
+        new token
+    """
+
+    helping_data: tp.Any = {
+        'refresh': {
+            'expire': REFRESH_TOKEN_EXPIRE_MINUTES,
+            'default_expire': DEFAULT_REFRESH_TOKEN_EXPIRE_MINUTES,
+            'subject': REFRESH_TOKEN_JWT_SUBJECT
+        },
+        'access': {
+            'expire': ACCESS_TOKEN_EXPIRE_MINUTES,
+            'default_expire': DEFAULT_ACCESS_TOKEN_EXPIRE_MINUTES,
+            'subject': ACCESS_TOKEN_JWT_SUBJECT
+        }
+    }
+
+    if helping_data[token_type]['expire'] is not None:
+        helping_data['expire'] = datetime.utcnow() + timedelta(minutes=float(helping_data[token_type]['expire']))
+    else:
+        helping_data['expire'] = datetime.utcnow() + timedelta(
+            minutes=float(helping_data[token_type]['default_expire'])
+        )
+
+    data.update({'exp': helping_data['expire'], 'sub': helping_data[token_type]['subject']})
+    encoded_jwt: str = jwt.encode(payload=data, key=SECRET_KEY, algorithm=TOKEN_ALGORITHM)
+    return encoded_jwt
+
+
+def generate_access_token_from_refresh(refresh_token: str) -> tp.Tuple[tp.Dict[str, str], int]:
+    """
+    Generate access token from refresh token
+    Args:
+        refresh_token: refresh_token
+    Return:
+        dict with access token and current status code
+    """
+
+    try:
+        payload = jwt.decode(refresh_token, SECRET_KEY, algorithms=[TOKEN_ALGORITHM])
+
+        if payload['sub'] != 'refresh':
+            return {'error': 'Refresh token is expected'}, 401
+
+        access_token = create_token(token_type='access',
+                                    data={
+                                        'username': payload["username"],
+                                        'password': payload["password"]
+                                    })
+        return {'access_token': access_token}, 200
+    except Exception:
+        return {'error': 'Invalid token or token is expired'}, 401
+
+
+def authenticate(func: tp.Any) -> tp.Any:
+    """
+    Decorator for authenticate user(set current user to request)
+    """
+
+    from flask import request, jsonify
+    from account.models import Account
+
+    @wraps(func)
+    def wrapper(*args: tp.Any, **kwargs: tp.Any) -> tp.Any:
+        if request.headers.get('Authorization') is None:
+            return jsonify({'error': 'Login required'})
+
+        token_components = request.headers.get("Authorization").split(' ')  # type: ignore
+        token_type = token_components[0]
+        access_token = token_components[-1]
+
+        if token_type != TOKEN_TYPE:
+            return jsonify({'error': 'Invalid token type'})
+
+        try:
+            payload = jwt.decode(access_token, SECRET_KEY, algorithms=[TOKEN_ALGORITHM])
+            if payload['sub'] != 'access':
+                return {'error': 'Access token is expected'}, 401
+            account = Account.query.filter_by(username=payload['username']).first()
+            request.user = account  # type: ignore
```

Comment:
why you use both user and account words? you can choose one and use it everywhere

Reply:
didnt understand what you were talking about, explain please

Reply:
what is the difference between user and account?